src/Features/CSharp/Portable/AddImport/CSharpAddMissingImportsFeatureService.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.AddImport;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CSharp.AddImport
+{
+    [ExportLanguageService(typeof(IAddMissingImportsFeatureService), LanguageNames.CSharp), Shared]
+    internal class CSharpAddMissingImportsFeatureService : AbstractAddMissingImportsFeatureService
+    {
+        [ImportingConstructor]
+        public CSharpAddMissingImportsFeatureService(IDiagnosticService diagnosticService)
+            :base(diagnosticService)
+        {
+        }
+
+        protected sealed override AbstractAddImportCodeFixProvider GetAddImportCodeFixProvider()
+        {
+            return new CSharpAddImportCodeFixProvider();
```

Comment:
this feels little bit backward. I would expect code fix provider using the service not the service using the code fix provider.

I would put log in the service and the codefix porivder just calling method in the service.